pkg/xray/watches.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	return *resource
+}
+
+func unpackFilters(list []interface{}) []WatchFilter {
+	var filters []WatchFilter
+
+	for _, raw := range list {
+		filter := new(WatchFilter)
+		f := raw.(map[string]interface{})
+		filter.Type = f["type"].(string) // TODO: recognize the type of the filter
+		filter.Value = f["value"].(string)
+		filters = append(filters, *filter)
+	}
+
+	return filters
+}
+
+func unpackAssignedPolicy(rawCfg interface{}) WatchAssignedPolicy {
+	policy := new(WatchAssignedPolicy)
+
+	cfg := rawCfg.(map[string]interface{})
+	policy.Name = cfg["name"].(string)
+	policy.Type = cfg["type"].(string)
+
+	return *policy
+}
+
+func packProjectResources(resources WatchProjectResources) []interface{} {
+	if len(resources.Resources) == 0 {
+		return []interface{}{}
+	}
+
+	var list []interface{}
+	for _, res := range resources.Resources {
+		resourceMap := map[string]interface{}{}
+		resourceMap["type"] = res.Type
+		if len(res.Name) > 0 {
+			resourceMap["name"] = res.Name
+		}
+		if len(res.BinaryManagerId) > 0 {
+			resourceMap["bin_mgr_id"] = res.BinaryManagerId
+		}
+		resourceMap["filter"] = packFilters(res.Filters)
+		list = append(list, resourceMap)
+	}
+
+	return list
+}
+
+func packFilters(filters []WatchFilter) []interface{} {
+	if filters == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, f := range filters {
+		m := map[string]interface{}{
+			"type":  f.Type,
+			"value": f.Value,
+		}
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func packAssignedPolicies(policies []WatchAssignedPolicy) []interface{} {
+	if policies == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, p := range policies {
+		m := make(map[string]interface{})
+		m["name"] = p.Name
+		m["type"] = p.Type
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func resourceXrayWatchCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	_, err := m.(*resty.Client).R().SetBody(watch).Post("xray/api/v2/watches")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(watch.GeneralData.Name) // ID may be returned according to the API docs, but not in go-xray
+	resourceXrayWatchRead(ctx, d, m)
+	return diags
+}
+
+func resourceXrayWatchRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	resp, err := m.(*resty.Client).R().SetResult(&watch).Get("xray/api/v2/watches/" + d.Id())
+
+	if err != nil {
+		if resp != nil && resp.StatusCode() == http.StatusNotFound {
+			log.Printf("[WARN] Xray watch (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diags
+	}
+	return diags
+}
+
+func resourceXrayWatchUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	_, err := m.(*resty.Client).R().SetBody(watch).Put("xray/api/v2/watches/" + d.Id())
```

Comment:
Shouldn't you do like you did in 207?

Reply:
not sure, it's legacy code. I'd keep it as is. Should the Update method set ID to empty, if it can't update the resource in general? 
The same logic is used in the [Artifactory provider](https://github.com/jfrog/terraform-provider-artifactory/blob/a0d2e8eddfd79e68f6304845e13efe556b29bc5a/pkg/artifactory/repositories.go#L176).

Reply:
With terraform, if the resource has been deleted outside of TF knowledge, then you set the ID to "" to indicate this. It's done in a few (in consistent) places. Remember, this is a completely new implementation - "legacy" Would assume you can make changes. 

if the resource is 404, set id to `""`

Reply:
Thanks! Added